src/katgpucbf/xbgpu/bsend.py

```diff
@@ -327,13 +326,16 @@ def enable_substream(self, stream_id: int, enable: bool = True) -> None:
     async def get_free_chunk(self) -> Chunk:
         """Obtain a :class:`.Chunk` for transmission.
 
-        If the chunk's :attr:`future` is cancelled
+        Raises
+        ------
+        asyncio.CancelledError
+            If the chunk's send future is cancelled.
         """
         chunk = await self._chunks_queue.get()
         try:
             await chunk.future
         except asyncio.CancelledError:
-            pass
+            raise asyncio.CancelledError("Sending chunk was cancelled")
         except Exception:
             logger.exception("Error sending chunk")
```

Comment:
This on the other hand suppresses any other exceptions.
Are you sure this is what you want to do?

Reply:
Ah, well. What I actually (and usually) want to do is use the [traceback](https://docs.python.org/3/library/traceback.html) module, but that might be some outdated idea/expectation I have about debugging.

OTOH, what can I/we actually do here? 
* I'm genuinely not sure if a hard failure would cause this to execute, and whether we'd want to stop the BPipeline altogether if this did happen. Genuinely, open to suggestions.
* Perhaps I could at least `chunk.future.set_result(None)` and send it back on its way?